From bb78531a60bfa75f269edd4f152c3e2a6e956e6a Mon Sep 17 00:00:00 2001
From: Jun Nie <jun.nie@linaro.org>
Date: Fri, 27 Jul 2018 09:15:15 +0800
Subject: [PATCH 2/2] rpi3: Enable boot kernel from fit image

Enable boot kernel from fit image with select configs
and specify boot script image node in fit image, bootscr
node in mbl case.

Code that reusing dtb in firmware is disabled.

Signed-off-by: Jun Nie <jun.nie@linaro.org>
---
 configs/rpi_3_32b_defconfig | 5 +++--
 include/configs/rpi.h       | 8 ++++++--
 2 files changed, 9 insertions(+), 4 deletions(-)

diff --git a/configs/rpi_3_32b_defconfig b/configs/rpi_3_32b_defconfig
index c00b4bb..247236b 100644
--- a/configs/rpi_3_32b_defconfig
+++ b/configs/rpi_3_32b_defconfig
@@ -5,6 +5,9 @@ CONFIG_TARGET_RPI_3_32B=y
 CONFIG_SYS_MALLOC_F_LEN=0x2000
 CONFIG_DEFAULT_DEVICE_TREE="bcm2837-rpi-3-b"
 CONFIG_DISTRO_DEFAULTS=y
+CONFIG_FIT=y
+CONFIG_FIT_SIGNATURE=y
+CONFIG_FIT_VERBOSE=y
 CONFIG_OF_BOARD_SETUP=y
 # CONFIG_DISPLAY_CPUINFO is not set
 # CONFIG_DISPLAY_BOARDINFO is not set
@@ -38,4 +40,3 @@ CONFIG_DM_VIDEO=y
 CONFIG_SYS_WHITE_ON_BLACK=y
 CONFIG_CONSOLE_SCROLL_LINES=10
 CONFIG_PHYS_TO_BUS=y
-CONFIG_OF_LIBFDT_OVERLAY=y
diff --git a/include/configs/rpi.h b/include/configs/rpi.h
index a97550b..2ac4373 100644
--- a/include/configs/rpi.h
+++ b/include/configs/rpi.h
@@ -15,6 +15,7 @@
 
 /* Architecture, CPU, etc.*/
 #define CONFIG_ARCH_CPU_INIT
+#define CONFIG_GZIP_COMPRESSED
 
 /* Use SoC timer for AArch32, but architected timer for AArch64 */
 #ifndef CONFIG_ARM64
@@ -159,7 +160,10 @@
 	"dhcpuboot=usb start; dhcp u-boot.uimg; bootm\0" \
 	ENV_DEVICE_SETTINGS \
 	ENV_MEM_LAYOUT_SETTINGS \
-	BOOTENV
-
+	BOOTENV \
+	"boot_a_script="                                                  \
+		"load ${devtype} ${devnum}:${distro_bootpart} "           \
+			"${scriptaddr} ${prefix}${script}; "              \
+		"source ${scriptaddr}:bootscr\0"
 
 #endif
-- 
2.7.4

